requirements.txt

```diff
@@ -1,39 +1,55 @@
+apiclient==1.0.4
 aspy.yaml==1.3.0
 atomicwrites==1.3.0
 attrs==19.3.0
 bleach==3.1.0
 build-utils==0.3.2
+cachetools==3.1.1
 certifi==2019.9.11
 cfgv==2.0.1
 chardet==3.0.4
 colorama==0.4.1
 docutils==0.15.2
 fake-rpi==0.6.2
+google-api-python-client==1.7.11
+google-auth==1.7.0
+google-auth-httplib2==0.0.3
+http-client==0.1.22
+httplib2==0.14.0
 identify==1.4.7
 idna==2.8
 importlib-metadata==0.23
 more-itertools==7.2.0
 nodeenv==1.3.3
 nose==1.3.7
 numpy==1.17.3
+oauth2client==4.1.3
 packaging==19.2
 paho-mqtt==1.4.0
+picamera==1.13
+pkg-resources==0.0.0
```

Comment:
Remove `pkg-resources`

Reply:
I will get on it soon.

Reply:
Okay the change was made.

Reply:
Please remove `picamera` as it cannot be installed by Travis's virtual machine. Keep an eye for these packages that usually fail Travis build, e.g. `picamera`, `RPi.GPIO`, `pkg-resources`, etc. I think at the current stage of our project, doing `pip freeze > requirements.txt` probably is no longer a recommended practice.

Reply:
My suggestion is first sync your `requirements.txt` with master. Then just add these packages manually `httplib2`, `google-api-python-client`, `oauth2client`. All the other packages are dependencies for these three and will be installed automatically. So there is no need to explicitly add them to `requirements.txt`

Reply:
Of course, if adding just these three packages is not enough, you can manually add more if needed.

Reply:
So, remove all the new packages added to requirements.txt besides httplib2, google-api-python-client, oauth2client? Or just remove picamera?

Reply:
try remove all the new packages except httplib2, google-api-python-client, oauth2client. Then delete your local venv, recreate it, and install with requirements. this is to see if using just these three would be sufficient for your camera interface,

Reply:
Did what you asked.

Reply:
`test_five` in `test_camera_interface` failed. Here is the error message.

```
_____________________________ TestClass.test_five ______________________________
self = <src.raspberry_pi_camera.tests.test_camera_interface.TestClass object at 0x7f92002c13c8>
    def test_five(self):
        capture = str(sample)
>       assert (
            capture
            == "Video Length: 30, Resolution: (640, 480), Save Location: /home/pi/Videos/"
        )
E       AssertionError: assert 'Video Length...JNiyHgG02ECXN' == 'Video Length...me/pi/Videos/'
E         Skipping 62 identical leading characters in diff, use -v to show
E         - /pi/Videos/, YouTube Livestream Link: https://youtu.be/t48LW4J8b4A, YouTube Playlist Link: https://www.youtube.com/playlist?list=PLTdMMnsiEwSnKNWdLlAEJNiyHgG02ECXN
E         + /pi/Videos/
src/raspberry_pi_camera/tests/test_camera_interface.py:33: AssertionError
========================= 1 failed, 17 passed in 5.57s =========================
The command "python3 -m pytest --ignore=src/raspberry_pi_ui/tests" exited with 1.
```
Seems to me that your capture now returns actual youtube link which is not included in the assertion check.

Has your `pytest` passed locally?

Reply:
I checked your branch and saw that you changed the return value of `__str__()` of your class. Yet such change was not reflected in your test file. So you need to update your test case.

Reply:
My bad, made the change.